README.adoc

```diff
@@ -59,28 +78,87 @@ module "backup" {
+----
+# Restore from a backup
+velero restore create --from-backup my-new-backup
+# Restore from the last backup of a schedule
+velero restore create --from-schedule restic-schedule
+----
+
+More options are described in the https://velero.io/docs/main/restore-reference/[Velero documentation].
+
+NOTE: In order to perform restore successfully, Velero need that the object doesn't already exists in the target cluster.
+As a disaster recovery tool, it is not able to merge an existing object such as a PVC with its backup.
+
+== Monitoring
+
+If enabled, this module exposes Velero metrics and setup a scrape target for Prometheus.
+
+=== Grafana dashboard
+
+This module contains a Grafana dashboard to monitor the frequency, state and frequency of backups, as shown here below.
+
+image:https://raw.githubusercontent.com/camptocamp/devops-stack-module-velero/doc/docs/modules/ROOT/assets/images/grafana-velero-dashboard.png["Grafana dashboard with backup informations",link=https://raw.githubusercontent.com/camptocamp/devops-stack-module-velero/doc/docs/modules/ROOT/assets/images/grafana-velero-dashboard.png]
+
+=== Alerts
+
+This module also setup some alerting in Prometheus for the following cases:
 
-_This section will contain the more technical details of the module. It is mainly generated automatically by Terraform Docs (everything that is between the comments `BEGIN_TF_DOCS` / `END_TF_DOCS`). However, it is recommended to, at the bare minimum, add a dependencies section to explain the reason on why this module could depend on another one from the DevOps Stack._
+- Partially failed backups ratio higher than a specified amount (`alert_partially_failed_ratio`)
+- Failed backups ratio higher than a specified amount (`alert_failed_ratio`)
+- No successful backup for a specified amount of time (`alert_backup_timeout`)
 
-_You can safely delete the content between the comments `BEGIN_TF_DOCS` / `END_TF_DOCS` and `BEGIN_TF_TABLES` / `END_TF_TABLES` when creating your on module._
+
+== Limitations
+
+=== KinD setup
+
+Velero cannot perform backups of `hostPath` volumes, which are the volumes used in a KinD setup. Therefore, this module cannot be used in a Devops stack on KinD.
+
+== Technical Reference
 
 === Dependencies
 
 ==== `module.kube-prometheus-stack`
 
-As this module contains a Prometheus metrics endpoint and a Grafana dashboard, it has to be deployed after the Prometheus stack.
+As this module contains a Prometheus metrics endpoint and a Grafana dashboard, it has to be deployed after the https://github.com/camptocamp/devops-stack-module-kube-prometheus-stack[Prometheus stack].
+
+==== `module.argocd`
 
-==== ``
+Velero is deployed as an ArgoCD application, which means that ArgoCD has to be deployed first.
 
+==== S3 bucket or equivalent
+
+In order to perform backups, Velero needs a storage location for backups, either a S3 compatible storage (AWS S3, Minio, etc.), Azure Blob storage, etc.
+
+Velero backs all Kubernetes objects to the designated
```

Comment:
I would put this block somewhere else, we can discuss together if you want.

The `Dependencies` section was designed more thinking about which other Terraform modules this module depends on.